### PR TITLE
Create API for register gym and test it

### PR DESCRIPTION
**Pull request by AhmedQeshta**
## Create API for register gym and test it

### CODE REVIEW

**Review by rawaneyad — APPROVED** [review]
Great Job Ahmed 👍🏼

File: server/utils/hashing.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { hash, compare } from 'bcryptjs';
+
+export function hashPassword(password: string): Promise<string> {
+  return hash(password, 10);
+}
+
+export function comparePassword(password: string, hashedPassword: string): Promise<boolean> {
+  return compare(password, hashedPassword);
```
Comment (LinaYahya) [inline server/utils/hashing.ts:8]: No need for this file, All you need to do when compare is to call `bcrypt` compare method, So Why to have a separate function?

File: server/utils/validation/gymRegisterSchema.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+const gymRegisterSchema = Joi.object({
+  gymName: Joi.string().required(),
+  logo: Joi.string().required(),
+  email: Joi.string()
+    .max(250)
+    .min(4)
+    .email({ minDomainSegments: 2, tlds: { allow: ['com', 'net'] } })
+    .required(),
+  password: Joi.string().max(50).min(3).required(),
```
Comment (LinaYahya) [inline server/utils/validation/gymRegisterSchema.ts:11]:
```suggestion
  password: Joi.string().max(15).min(8).required(),
```

File: server/utils/aws/index.ts [hunk 3]
```diff
@@ -7,18 +7,37 @@ const s3 = new AWS.S3({
...
   return s3.upload(params).promise();
 };
-export default uploadImage;
+
+interface deleteParamsType {
+  Bucket: string;
+  Key: string;
+}
+
+function deleteImage(Key: string): Promise<any> {
```
Comment (LinaYahya) [inline server/utils/aws/index.ts:30]: what is the purpose to create `deleteImage` function in signup?
Remove this function from here

File: server/controllers/auth/gymRegister.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      });
+  } catch (error: any) {
+    console.log(error);
+
+    if (error.name === 'ValidationError') {
+      let errors: Array<string> = [];
+      error.details.map((detail: any) => {
+        errors = [...errors, detail.message];
+      });
+      return next(new CustomError(errors.toString(), 400));
```
Comment (LinaYahya) [inline server/controllers/auth/gymRegister.ts:88]: as you need to convert array to string, create it from start as a string,
Not sure If we're going to display it to user,
But I guess we can start with empty string then `error \n nextError`, So in the map itself add the new error beside `\n`, and reduce would be a better choice then

Alsom as we going to use it every where create a function to take errors from validation and return the output we want

File: server/controllers/auth/gymRegister.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        maxAge: 900000,
+        httpOnly: true,
+      })
+      .json({
+        status: 201,
+        message: 'تم تسجيل النادي بنجاح',
+        payload,
+      });
+  } catch (error: any) {
+    console.log(error);
```
Comment (LinaYahya) [inline server/controllers/auth/gymRegister.ts:81]:
```suggestion
```

File: .eslintrc.json [hunk 6]
```diff
@@ -24,7 +25,8 @@
     "no-console": "off",
     "object-curly-newline": "off",
     "operator-linebreak": "off",
-    "no-return-await": "off"
+    "no-return-await": "off",
+    "array-callback-return": "off"
```
Comment (LinaYahya) [inline .eslintrc.json:29]:
```suggestion
   
```

File: server/controllers/auth/gymRegister.ts [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    // Generate payload
+    const payload = {
+      id: gym.getDataValue('id'),
+      name: gym.getDataValue('gymName'),
+      role: 'gym',
+    };
+    // Generate the token
+    const token = await generateToken(payload);
+
+    return res
```
Comment (LinaYahya) [inline server/controllers/auth/gymRegister.ts:69]: No need to return here

File: server/controllers/auth/gymRegister.ts [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+      typeGender,
+      monthlyPrice,
+      sixMonthPrice,
+      fulltime,
+      features,
+    });
+
+    // Generate payload
+    const payload = {
+      id: gym.getDataValue('id'),
```
Comment (LinaYahya) [inline server/controllers/auth/gymRegister.ts:62]:
```suggestion
      id: gym.id,
```

**Review by LinaYahya — COMMENTED** [review]
Good job Ahmed, small changes to fix,
Also add JWT secret in the `yml` file, to get from repo secrets

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**